cranelift/codegen/src/isa/aarch64/pcc.rs

```diff
@@ -254,6 +279,44 @@ pub(crate) fn check(
             }
         }
 
+        Inst::CSel { rd, cond, rn, rm } if cond == Cond::Hs && cmp_flags.is_some() => {
+            let (cmp_lhs, cmp_rhs) = cmp_flags.unwrap();
+            trace!("CSel: cmp {cond:?} ({cmp_lhs:?}, {cmp_rhs:?})");
+
+            check_output(ctx, vcode, rd, &[], |vcode| {
+                // We support transitivity-based reasoning. If the
+                // comparison establishes that
+                //
+                //   (x+K1) <= (y+K2)
+                //
+                // then on the true-side of the select we can edit the maximum
+                // in a DynamicMem or DynamicRange by replacing x's with y's
+                // with appropriate offsets -- this widens the range.
+                //
+                // Likewise, on the false-side of the select we can
+                // replace y's with x's -- this also widens the range. On
+                // the false side we know the inequality is strict, so we
+                // can offset by one.
+
+                // True side: lhs <= rhs, not strict.
+                let rn = get_fact_or_default(vcode, rn, 64);
+                trace!("rn = {rn:?}");
+                let rn = ctx.apply_inequality(&rn, &cmp_lhs, &cmp_rhs, false);
+                trace!(" -> rn = {rn:?}");
+                // false side: rhs < lhs, strict.
+                let rm = get_fact_or_default(vcode, rm, 64);
+                trace!("rm = {rm:?}");
+                let rm = ctx.apply_inequality(&rm, &cmp_rhs, &cmp_lhs, true);
+                trace!(" -> rm = {rm:?}");
+
+                let union = Fact::union(&rn, &rm);
+                trace!("union = {union:?}");
```

Comment:
Let's either clean up these traces and give them more context to make them readable by general wasmtime/cranelift devs or remove them.

Reply:
Done